partiql-planner/src/main/kotlin/org/partiql/planner/internal/transforms/NormalizeFromSource.kt

```diff
@@ -32,19 +32,32 @@ import org.partiql.planner.internal.util.BinderUtils.toBinder
  * Assign aliases to any FROM source which does not have one.
  */
 internal object NormalizeFromSource : AstPass {
+    override fun apply(statement: Statement): Statement = statement.accept(Visitor, IntCounter(0)) as Statement
 
-    override fun apply(statement: Statement): Statement = statement.accept(Visitor, 0) as Statement
-
-    private object Visitor : AstRewriter<Int>() {
+    internal class IntCounter(var counter: Int = 0) {
+        fun next(): Int = counter++
+    }
 
+    private object Visitor : AstRewriter<IntCounter>() {
         // Each SFW starts the ctx count again.
-        override fun visitQueryBodySFW(node: QueryBody.SFW, ctx: Int): AstNode = super.visitQueryBodySFW(node, 0)
+        override fun visitQueryBodySFW(node: QueryBody.SFW, ctx: IntCounter): AstNode =
+            super.visitQueryBodySFW(node, IntCounter(0))
 
-        override fun visitFrom(node: From, ctx: Int) = super.visitFrom(node, ctx) as From
+        override fun visitFrom(node: From, ctx: IntCounter): From {
+            // Handle the case where From contains multiple table references
+            if (node.tableRefs.size > 1) {
+                val newTableRefs = node.tableRefs.map { tableRef ->
+                    tableRef.accept(this, ctx) as FromTableRef // assigning different context values to each table reference
+                }
+                return From(newTableRefs)
+            }
+            // Handle single table
+            return super.visitFrom(node, ctx) as From
+        }
```

Comment:
With the context changes, I believe we can actually get rid of this overridden visit function.